File: examples/scala/psenvsub.scala
/*
 * Pubsub envelope subscriber
 *
 * @author Giovanni Ruggiero
 * @email giovanni.ruggiero@gmail.com
 */

import org.zeromq.ZMQ

object psenvsub {
  def main(args : Array[String]) {

		// Prepare our context and subscriber
		val context = ZMQ.context(1)
		val subscriber = context.socket(ZMQ.SUB)

		subscriber.connect("tcp://localhost:5563")
		subscriber.subscribe("B".getBytes())
		while (true) {
			// Read envelope with address
			val address = new String(subscriber.recv(0))
			// Read message contents
			val contents = new String(subscriber.recv(0))
			println(address + " : " + contents)
		}
	}
}

File: examples/Java/Psenvsub.java
import org.zeromq.ZMQ;
import org.zeromq.ZMQ.Context;
import org.zeromq.ZMQ.Socket;

/**
 * Pubsub envelope subscriber
 */

public class psenvsub {

    public static void main (String[] args) {

        // Prepare our context and subscriber
        Context context = ZMQ.context(1);
        Socket subscriber = context.socket(ZMQ.SUB);

        subscriber.connect("tcp://localhost:5563");
        subscriber.subscribe("B".getBytes());
        while (!Thread.currentThread ().isInterrupted ()) {
            // Read envelope with address
            String address = subscriber.recvStr ();
            // Read message contents
            String contents = subscriber.recvStr ();
            System.out.println(address + " : " + contents);
        }
        subscriber.close ();
        context.term ();
    }
}
